src/pykeen/models/unimodal/compgcn.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of the Comp-GCN model."""
+
+from typing import Any, Mapping, Optional
+
+import torch
+from class_resolver import Hint
+
+from ..nbase import ERModel
+from ...nn import EmbeddingSpecification
+from ...nn.emb import CompGCNRepresentation, SingleCompGCNRepresentation
+from ...nn.modules import DistMultInteraction, Interaction, interaction_resolver
+from ...triples import TriplesFactory
+from ...typing import RelationRepresentation
+
+__all__ = [
+    "CompGCN",
+]
+
+
+class CompGCN(
+    ERModel[torch.FloatTensor, RelationRepresentation, torch.FloatTensor],
+):
+    """An implementation of CompGCN from [vashishth2020]_.
+
+    This model uses graph convolutions, and composition functions.
+
+    ---
+    citation:
+        author: Vashishth
+        year: 2020
+        link: https://arxiv.org/pdf/1911.03082
+        github: malllabiisc/CompGCN
+    """
+
+    #: The default strategy for optimizing the model"s hyper-parameters
+    hpo_default = dict(
+        embedding_dim=dict(type=int, low=32, high=512, q=32),
+    )
+
+    def __init__(
+        self,
+        *,
+        triples_factory: TriplesFactory,
+        encoder_kwargs: Optional[Mapping[str, Any]] = None,
+        interaction: Hint[Interaction[torch.FloatTensor, RelationRepresentation, torch.FloatTensor]] = DistMultInteraction,
+        interaction_kwargs: Optional[Mapping[str, Any]] = None,
+        **kwargs,
+    ):
+        encoder_kwargs = encoder_kwargs or {}
+
+        # TODO: Remove this, once testing is updated
+        embedding_dim = kwargs.pop("embedding_dim", None)
```

Comment:
@cthoyt 

This was necessary for testing, since we do not (yet) use `unittest_template` with `_pre_instantiation_hook` to pop the `embedding_dim` from the initializer's kwargs.

Reply:
ooooo that sounds like fun. is there any reason we can't include that update in this PR?

Reply:
It is not necessarily related to the CompGCN model. :thinking:

Reply:
@mberr now that the other PR is merged, I think you can update this